### PR TITLE
fix: Check the `actions` field in flows when running a trace

### LINKED ISSUE #61: Internal Server Error when running a trace over a Flow without actions - drop rule
Hi,

When running Sdntrace in a switch that has a Flow without actions results in Internal Server Error, as shown below. Flows without actions are valid entries, which means a drop rule.

1. Create a flow for drop packets (no actions):
```
curl -X POST -H 'Content-type: application/json' -d '{"flows": [{"priority": 10, "match": {"in_port": 60, "dl_vlan": 777}}]}' http://127.0.0.1:8181/api/kytos/flow_manager/v2/flows/00:00:00:00:00:00:00:11
```

2. Make sure the flow was created:
```
# ovs-ofctl dump-flows Ampath1 dl_vlan=777
 cookie=0x0, duration=474.623s, table=0, n_packets=0, n_bytes=0, priority=10,in_port=60,dl_vlan=777 actions=drop
```

3. Run a trace:
```
# curl -H 'Content-type: application/json' -X PUT http://127.0.0.1:8181/api/amlight/sdntrace_cp/trace -d '{"trace": {"switch": {"dpid": "00:00:00:00:00:00:00:11", "in_port": 60}, "eth": {"dl_type": 33024, "dl_vlan": 777}}}'
{"code":500,"description":"The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.","name":"Internal Server Error"}

# tail -n1 /var/log/syslog
Jan 20 19:20:54 5003be03af14 kytos.core.controller:ERROR app:1449:  Exception on /api/amlight/sdntrace_cp/trace [PUT]#012Traceback (most recent call last):#012  File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 2073, in wsgi_app#012    response = self.full_dispatch_request()#012  File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 1519, in full_dispatch_request#012    rv = self.handle_user_exception(e)#012  File "/usr/local/lib/python3.9/dist-packages/flask_cors/extension.py", line 165, in wrapped_function#012    return cors_after_request(app.make_response(f(*args, **kwargs)))#012  File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 1517, in full_dispatch_request#012    rv = self.dispatch_request()#012  File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 1503, in dispatch_request#012    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)#012  File "//var/lib/kytos/napps/amlight/sdntrace_cp/main.py", line 64, in trace#012    result = self.tracepath(entries, stored_flows)#012  File "//var/lib/kytos/napps/amlight/sdntrace_cp/main.py", line 105, in tracepath#012    result = self.trace_step(switch, entries, stored_flows)#012  File "//var/lib/kytos/napps/amlight/sdntrace_cp/main.py", line 146, in trace_step#012    flow, entries, port = self.match_and_apply(#012  File "//var/lib/kytos/napps/amlight/sdntrace_cp/main.py", line 235, in match_and_apply#012    actions = flow['flow']['actions']#012KeyError: 'actions'
```

Expected behavior: one entry with the starting trace step and `"out": None` (no output from that flow -- it could be something else as well like `"out": "drop"`)

### PR DESCRIPTION
**Pull request by gretelliz**
Closes #61 

### Summary

Now, it is checked if `actions` is a field of the flow [See link](https://github.com/kytos-ng/sdntrace_cp/blob/86b466560b449b20028512ad4aa1f632bff8946e/main.py#L238).

### Local Tests

Now, when running Sdntrace in a switch that has a Flow without actions:

Create a flow for drop packets (no actions):

```
curl -X POST -H 'Content-type: application/json' -d '{"flows": [{"priority": 10, "match": {"in_port": 60, "dl_vlan": 777}}]}' http://127.0.0.1:8181/api/kytos/flow_manager/v2/flows/00:00:00:00:00:00:00:01
```

Run a trace:

```
curl -H 'Content-type: application/json' -X PUT http://127.0.0.1:8181/api/amlight/sdntrace_cp/trace -d '{"trace": {"switch": {"dpid": "00:00:00:00:00:00:00:01", "in_port": 60}, "eth": {"dl_type": 33024, "dl_vlan": 777}}}
```

Response:

```
{"result":[{"dpid":"00:00:00:00:00:00:00:01","out":null,"port":60,"time":"2023-02-06 13:27:59.390535","type":"starting","vlan":777}]}
```

### CODE REVIEW

**Review by ajoaoff — COMMENTED** [review]
@gretelliz It looks fine, but could you please add a test with a stored flow with no actions?

**Comment by gretelliz** [comment]
> @gretelliz It looks fine, but could you please add a test with a stored flow with no actions?

@ajoaoff, thank you. Sure, I'll do it.

**Review by ajoaoff — APPROVED** [review]
LGTM

**Review by viniarck — CHANGES_REQUESTED** [review]
LGTM, but please also update the changelog file. 

Also, let's add an e2e test with an empty action (drop) in topology with at least three switches:

1 S1 2 ---- 1 S2 2 ---- 1 S3

and then on SW1 you push a flow matching port 1 and output port 2, but then on S2 you match on port 1 and empty actions, let's ensure that the output of sdntrace_cp is correctly showing that the trace doesn't complete.

**Comment by viniarck** [comment]
@gretelliz thanks for updating the changelog, you ended up adding the information on section `2022.3.1`, but this change landing on `master` is considered unreleased. Can you fix/move it please?

**Comment by gretelliz** [comment]
@viniarck, You are right, my bad.

**Comment by viniarck** [comment]
> @viniarck, You are right, my bad.

Looks good now, feel free to merge this once the e2e is also ready to land. Thanks.

**Pull request merged**
